### PR TITLE
Reorder core functions matcher

### PR DESCRIPTION
**Pull request by azemoh**
Problem: Not highlighting core functions that start with `to_*` properly.
old

Fix: Reorder matcher so `to` comes after affected functions.
new

- [x] Build pass!
- [x] Follow VS Code [Coding Guidelines](https://github.com/Microsoft/vscode/wiki/Coding-Guidelines)

### CODE REVIEW

File: syntaxes/ruby.tmLanguage [hunk 1]
```diff
@@ -3000,7 +3000,7 @@
...
...onnect|test_s_const_load|test_s_test_s_new_DCOM|test_s_new_from_clsid|test_s_ole_classes|test_s_progids|test_s_typelibs|test_setproperty|test_setproperty_bracket|test_setproperty_with_equal|test_src_type|test_to_s|test_typekind|test_value|test_variables|test_variant|test_visible|test|text_field|textarea|text|time_to_day_fraction|timeout|times|tmpdir|to_ary|to_a|to_enum|to_f|to_hash|to_int|to_io|to_i|to_mailtext|to_proc|to_rfc822text|to_r|to_set|to_str|to_sym|to_s|to|today|toeuc|tojis|tosjis|total_in|total_out|touch|toutf16|toutf8|tr_s|trace_var|trace|transaction|transfer|transformNode|transformNodeToObject|transpose|trap|truncate|try_implicit|try_lock|tr|tv_sec|tv_usec|typekind|typelibs|type|t|uid|umask|unbind|undef_method|undefine_finalizer|unescapeElement|unescapeHTML|unescape|ungetc|union|uniqueID|uniq|unknown|unlink|unlock|unpack|unscan|unshift|untaint|untrace_var|unused|upcase|update|upto|url_encode|url|use_registry|usec|userinfo|user|utc_offset|utc|utime|u|validateOnParse|values_at|values|value|variable_kind|variables|varkind|version|wait2|waitall|waitpid|waitpid2|wait|wakeup|warn|wday|wrap|write|xmlschema|xml|yday|year|yield|zero|zip|zlib_version|zone_offset|zone)\b</string>
```
Comment (HookyQR) [inline syntaxes/ruby.tmLanguage]: The first block doesn't have a word edge at the end. I would be easier to just make this a single group:
at `|zero\?)|\b(__getobj__` remove the `)|\b(` and replace with `|`.

Also, I think the `(?&lt;=\.)` (preceded by '.') is unnecessary in front of `line`.

Same for the next block.

Comment (azemoh) [inline syntaxes/ruby.tmLanguage]: Sure. What does this do?

**Comment by azemoh** [comment]
Sure, done.

File: syntaxes/ruby.tmLanguage [hunk 2]
```diff
@@ -3000,7 +3000,7 @@
...
...st_s_const_load|test_s_test_s_new_DCOM|test_s_new_from_clsid|test_s_ole_classes|test_s_progids|test_s_typelibs|test_setproperty|test_setproperty_bracket|test_setproperty_with_equal|test_src_type|test_to_s|test_typekind|test_value|test_variables|test_variant|test_visible|test|text_field|textarea|text|time_to_day_fraction|timeout|times|tmpdir|to_ary|to_a|to_enum|to_f|to_hash|to_int|to_io|to_i|to_mailtext|to_proc|to_rfc822text|to_r|to_set|to_str|to_sym|to_s|to|today|toeuc|tojis|tosjis|total_in|total_out|touch|toutf16|toutf8|tr_s|trace_var|trace|transaction|transfer|transformNode|transformNodeToObject|transpose|trap|truncate|try_implicit|try_lock|tr|tv_sec|tv_usec|typecode|typekind|typelibs|type|t|uid|umask|unbind|undef_method|undefine_finalizer|unescapeElement|unescapeHTML|unescape|ungetc|union|uniqueID|uniq|unknown|unlink|unlock|unpack|unscan|unshift|untaint|untrace_var|unused|upcase|update|upto|url_encode|url|use_registry|usec|userinfo|user|utc_offset|utc|utime|u|validateOnParse|values_at|values|value|variable_kind|variables|varkind|version|wait2|waitall|waitpid|waitpid2|wait|wakeup|warn|wday|wrap|write|xmlschema|xml|yday|year|yield|zero|zip|zlib_version|zone_offset|zone)\b</string>
```
Comment (HookyQR) [inline syntaxes/ruby.tmLanguage]: `fragment`, `host` `java_kind_of?` -> `java_kind_of\?`

In both.

**Comment by HookyQR** [comment]
Looks good. Waiting for CI (which shouldn't change for this as it's not tested.)

Happy to have this merged @rebornix.

**Pull request merged**